barretenberg/cpp/src/barretenberg/stdlib/primitives/databus/databus.hpp

```diff
@@ -77,54 +79,95 @@ template <class Builder> class DataBusDepot {
+     * calldata to K_{i}, and (2) that the return_data of K_{i-1} was calldata to K_{i}.
      *
-     * @param commitments Witness polynomial commitments for an key that has been accumulated
-     * @param public_inputs The public inputs of that key
-     * @param propagation_data Data about the presence of databus commitments on the public inputs of the key.
+     * @param return_data Return data from either an app or a kernel
+     * @param calldata Calldata corresponding to return data from a previous kernel
+     * @param secondary_calldata Calldata corresponding to some app return data
+     * @param public_inputs Public inputs of a kernel proof which contain propagated return data commitments
+     * @param propagation_data Info about the return data commitments stored in the provided public inputs
      */
-    void execute(const WitnessCommitments& commitments,
-                 const std::vector<Fr>& public_inputs,
-                 const DatabusPropagationData& propagation_data)
+    void set_return_data_to_be_propagated_and_perform_consistency_checks(const Commitment& return_data,
+                                                                         const Commitment& calldata,
+                                                                         const Commitment& secondary_calldata,
+                                                                         const std::vector<Fr>& public_inputs,
+                                                                         const DatabusPropagationData& propagation_data)
     {
-        // Flag indicating whether the input data corresponds to a kernel decider proving key (else, an app decider
-        // proving key). This is used to indicate whether the return data commitment being propagated belongs to a
-        // kernel or an app so that it can be checked against the appropriate calldata commitment in a subsequent round.
-        bool is_kernel_data = propagation_data.is_kernel;
-
-        // Assert equality between return data commitments propagated via the public inputs and the corresponding
-        // calldata commitment
-        if (propagation_data.contains_app_return_data_commitment) { // public inputs contain [R]_app
-            ASSERT(is_kernel_data); // Only kernels should contain databus commitments in their public inputs
-            size_t start_idx = propagation_data.app_return_data_public_input_idx;
-            Commitment app_return_data = reconstruct_commitment_from_public_inputs(public_inputs, start_idx);
-            // App return data should correspond to the secondary calldata of the subsequent kernel
-            assert_equality_of_commitments(app_return_data, commitments.secondary_calldata);
+        // Set the kernel/app return data commitment to be propagated via the public inputs
+        if (propagation_data.is_kernel) {
+            kernel_return_data_commitment = return_data;
```

Comment:
Not something that is a priority now but I wonder whether the `*_exists` variables  shouldn't be bool_t and the `kernel_return_data_commitment` be set via conditional assigns.

Reply:
Added a note of this to the larger issue for these default commitments